tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/CheckAllTool/ChangelogValidationTool.cs

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.Json;
+using Azure.Sdk.Tools.Cli.Services;
+using Azure.Sdk.Tools.Cli.Contract;
+using Azure.Sdk.Tools.Cli.Commands;
+using Azure.Sdk.Tools.Cli.Models;
+using Azure.Sdk.Tools.Cli.Helpers;
+using ModelContextProtocol.Server;
+
+namespace Azure.Sdk.Tools.Cli.Tools
+{
+    /// <summary>
+    /// This tool runs changelog validation for SDK projects.
+    /// </summary>
+    [Description("Run changelog validation for SDK projects")]
+    [McpServerToolType]
+    public class ChangelogValidationTool : MCPTool
+    {
+        private readonly ILogger<ChangelogValidationTool> logger;
+        private readonly IOutputService output;
+        private readonly IGitHelper gitHelper;
+
+        private readonly Option<string> packagePathOption = new(["--package-path", "-p"], "Path to the package directory to check") { IsRequired = true };
+
+        public ChangelogValidationTool(ILogger<ChangelogValidationTool> logger, IOutputService output, IGitHelper gitHelper) : base()
+        {
+            this.logger = logger;
+            this.output = output;
+            this.gitHelper = gitHelper;
+            CommandHierarchy = [SharedCommandGroups.Checks];
+        }
+
+        public override Command GetCommand()
+        {
+            Command command = new("changelog-validation", "Run changelog validation for SDK projects");
+            command.AddOption(packagePathOption);
+            command.SetHandler(async ctx => { await HandleCommand(ctx, ctx.GetCancellationToken()); });
+            return command;
+        }
+
+        public override async Task HandleCommand(InvocationContext ctx, CancellationToken ct)
+        {
+            var packagePath = ctx.ParseResult.GetValueForOption(packagePathOption);
+            var result = await RunChangelogValidation(packagePath);
+
+            output.Output(result);
+            ctx.ExitCode = ExitCode;
+        }
+
+        [McpServerTool(Name = "RunChangelogValidation"), Description("Run changelog validation for SDK packages. Provide absolute path to package root as param.")]
+        public async Task<ICLICheckResponse> RunChangelogValidation(string packagePath)
+        {
+            var stopwatch = System.Diagnostics.Stopwatch.StartNew();
```

Comment:
Note for me, we should auto stopwatch all tool calls if we can.

Reply:
This is something we can add as part of telemetry.